### PR TITLE
Move site visibility settings to a new tab

### LINKED ISSUE #46133: [Launch Your Store] Site Visibility tab
## Issue Description
Move the site visibility settings to a new tab named "Site Visibility".

Image

## Acceptance Criteria
* Create a settings tab named "Site Visibility"
* Move the existing site visibility section from the General tab to the new tab
* Update the link in the homescreen dropdown and front-end banner.

### PR DESCRIPTION
**Pull request by moon0326**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #46133  .

This PR moves site visibility settings to its own `Site Visibility` tab. This PR also fixes a bug where site visibility slotfill renders in the tax tab. 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with `launch-your-store` flag enabled.
2. Go to `WooCommerce -> Settings -> Site Visibility`
4. Change settings and click on `Save changes`
5. Confirm changes are saved as expected.
6. Go to `WooCommerce -> Home`
7. Click on the `Store coming soon` and then click on `Manage site visibility`
8. Confirm you're redirected to site visibility settings page.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Move site visibility settings to a new tab

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-site-visibility.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+			array(
+				array(
+					'id'   => 'wc_settings_general_site_visibility_slotfill',
+					'type' => 'slotfill_placeholder',
+				),
+			);
+
+		$settings = apply_filters( 'woocommerce_settings_pages', $settings );
+
+		if ( wc_site_is_https() ) {
```
Comment (adrianduffell) [inline plugins/woocommerce/includes/admin/settings/class-wc-settings-site-visibility.php]: Is the ssl handling needed here?

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-site-visibility.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	/**
+	 * Get settings for the default section.
+	 *
+	 * @return array
+	 */
+	protected function get_settings_for_default_section() {
+		$settings =
+			array(
+				array(
+					'id'   => 'wc_settings_general_site_visibility_slotfill',
```
Comment (adrianduffell) [inline plugins/woocommerce/includes/admin/settings/class-wc-settings-site-visibility.php]: Could `general` be removed from the id?

**Review by adrianduffell — CHANGES_REQUESTED** [review]
Thanks @moon0326! I tested on JN -- the tab worked, but the settings didn't appear on the screen 🤔 

I left a few minor questions as well.

Screen Shot 2024-04-03 at 9 33 13 am

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-site-visibility.php (hunk 1 above)
Comment (adrianduffell) [reply plugins/woocommerce/includes/admin/settings/class-wc-settings-site-visibility.php]: I see it's fixed in a new commit, thanks!

**Comment by adrianduffell** [comment]
> the tab worked, but the settings didn't appear on the screen 🤔

@moon0326 Oh, I didn't have the feature flag enabled 🤦 Once I fixed that the settings appeared on the screen. However, I think it means the tab code must need to be wrapped in a feature flag check?

**Comment by moon0326** [comment]
> it means the tab code must need to be wrapped in a feature flag check?

Ah I missed that! Thank you for testing that out 👍 I'll update the code.

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-site-visibility.php (hunk 2 above)
Comment (moon0326) [reply plugins/woocommerce/includes/admin/settings/class-wc-settings-site-visibility.php]: @adrianduffell Good catch!

File: plugins/woocommerce/tests/php/includes/settings/class-wc-settings-general-test.php [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<?php
+wj<?php
```
Comment (psealock) [inline plugins/woocommerce/tests/php/includes/settings/class-wc-settings-general-test.php]: typo?

**Review by psealock — APPROVED** [review]
This is testing well, just a minor comment. Pre-approving pending feedback from @adrianduffell

File: plugins/woocommerce/tests/php/includes/settings/class-wc-settings-general-test.php (hunk 3 above)
Comment (moon0326) [reply plugins/woocommerce/tests/php/includes/settings/class-wc-settings-general-test.php]: It is 😃 Thank you for catching it!

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-site-visibility.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+ * WC_Settings_Advanced.
+ */
+class WC_Settings_Site_Visibility extends WC_Settings_Page {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->id    = 'site-visibility';
+		$this->label = __( 'Site Visibility', 'woocommerce' );
```
Comment (adrianduffell) [inline plugins/woocommerce/includes/admin/settings/class-wc-settings-site-visibility.php]: I had the case wrong in the label. Could we please change it to: `Site visibility` (lowercase v)

File: plugins/woocommerce/tests/php/includes/settings/class-wc-settings-general-test.php (hunk 3 above)
Comment (moon0326) [reply plugins/woocommerce/tests/php/includes/settings/class-wc-settings-general-test.php]: @psealock Fixed in https://github.com/woocommerce/woocommerce/pull/46140/commits/e25d8d0a48d3c08f9c3a5be5bd18ddee728a551a

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-site-visibility.php (hunk 4 above)
Comment (moon0326) [reply plugins/woocommerce/includes/admin/settings/class-wc-settings-site-visibility.php]: @adrianduffell Updated in https://github.com/woocommerce/woocommerce/pull/46140/commits/48802c2b087833e349b005ef8350aeb44803c5ad

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-site-visibility.php (hunk 2 above)
Comment (moon0326) [reply plugins/woocommerce/includes/admin/settings/class-wc-settings-site-visibility.php]: @adrianduffell Updated in https://github.com/woocommerce/woocommerce/pull/46140/commits/086a3e5107b0e668e7a95ca18603577b9aa6e4a9

**Review by adrianduffell — APPROVED** [review]

**Pull request merged**